### PR TITLE
Banner, Indicator Number, Badge Improvements & Button Improvements 

### PR DESCRIPTION
**Pull request by hchamorro**
**Banner Updates:**
Add custom button
General styling improvements
Depreciate type 'Permanent' in favor of default behavior
**IconButton Indicator:**
Ability to display number in indicator
**Badge Improvements:**
less margin - can add more margin to UI as needed
pass icon
add size props
**Button:**
Add new size, xs
General Styling improvements

Created size xs for `Text`

### CODE REVIEW

File: packages/ui/src/Banner.tsx [hunk 1]
```diff
@@ -16,8 +17,16 @@ export const hideBanner = (id: string): Promise<void> => {
   return Unifier.storage.setItem(getKey(id), "true");
 };
 
+export const BannerType = {
+  ACTION: "action",
```
Comment (joshgachnang) [inline packages/ui/src/Banner.tsx:21]: What's the benefit of this over ensuring we're using typing properly?

**Review by joshgachnang — COMMENTED** [review]
Overall looks good! 

It's a little confusing to have both Badge and Chip. Could we update Badge to include a size ("sm" being the default and closer to what we have, "md" what you have here) and the icon prop instead? I'm down to make Badge more rounded like you have here.

File: packages/ui/src/Banner.tsx (hunk 1 above)
Comment (hchamorro) [reply packages/ui/src/Banner.tsx:21]: yep, I just don't like to type strings. Lemme know if you want me to remove

**Comment by hchamorro** [comment]
> Overall looks good!
> 
> It's a little confusing to have both Badge and Chip. Could we update Badge to include a size ("sm" being the default and closer to what we have, "md" what you have here) and the icon prop instead? I'm down to make Badge more rounded like you have here.

This is my bad. I created Chip in Flourish and just moved it over when doing these other changes. I hadn't thought to use badge and forgot we have it

File: packages/ui/src/Banner.tsx (hunk 1 above)
Comment (joshgachnang) [reply packages/ui/src/Banner.tsx:21]: Gotcha! I'm fine with this PR, but we should collectively decide on types vs these const/enums so we can have consistency.

**Review by joshgachnang — APPROVED** [review]

**Pull request merged**